README.md

```diff
@@ -2,43 +2,142 @@
+      - "NSXManagerService"
+      - "NSXControllerService"
+      - "SDDCHealth Instance"
+      - "vCenterBackupJob"
+```
+
+#### VCOPS (vCenter Operations) Adapter
+
+VCOPS resourcekinds can be defined in [inventory-config](tests/inventory_config.yaml):
+
+```yaml
+resourcekinds:
+    vcops_resourcekinds:
+      - "vC-Ops-Analytics"
+      - "vC-Ops-CaSA"
+      - "vC-Ops-Cluster"
+      - "vC-Ops-Collector"
+      - "vC-Ops-Node"
+      - "vC-Ops-Suite-API"
+      - "vC-Ops-Watchdog"
+```
+
+## Design
+
+For the REST queries of values, the resource-uuids (unique unit identifier) processed in the vROps must be used. For this purpose, the exporter was divided into two main components. 
+
+#### inventory
+
+The inventory collects all supported resourcekinds in their hierarchical relation, and makes them available at an internal API. The resourcekinds are updated by a continuous cycle that can be configured with `--sleep`.
+To have more control over the resources to be collected, they can be filtered by _resourcestatus_, _resourcehealth_ and _resourcestate_ in [inventory-config](tests/inventory_config.yaml).
+
+Multiple vROps can be processed concurrently. This is implemented with threads.
+
+###### inventory endpoints
+```shell
+GET
+
+/vrops_list                                       # list of one or multiple vrops
+/<target>/<resourcekind>/<int:iteration>          # path for each resourcekind
+/alertdefinitions/                                # vrops pre-defined alertdefinitions
+/iteration                                        # current inventory iteration
+/amount_resources                                 # amount of resources for each resourcekind
+/collection_times                                 # measured time for a inventory run per vrops
+/api_response_codes                               # HTTP response codes per resourcekind GET request
+/target_tokens                                    # dict with vrops: auth token
+```
+#### exporter
+
+The second component are the collectors that exist for each resourcekind as well as for metrics, properties and alerts. Each collector is responsible for only one task per resourcekind and one type of values. First, the resourcekinds in question are queried at the inventory's internal API. In the second step, the desired values, properties or alerts are queried. Each collector performs only one task - one resourcekind and one type from the three different values. First, the resourcekinds in question are queried at the inventory's internal API. In the second step, the values, properties or alarms are queried. From these, the Prometheus metrics are generated. To complete the picture, the metrics are enriched with the labels from the resourcekind relationships created in the inventory.
+
+![](images/architecture.png)
+
+To avoid multiple implementations of functionality, the collectors follow an inheritance structure.
+
+![](images/collectors.png)
+
+## Running the software
```

Comment:
Should we move this to the top? If not, we should give it a table of contents at the beginning of this documentation.